server/data/prisonApi.ts

```diff
@@ -133,4 +146,39 @@ export class PrisonApi extends RestClient {
       ),
     )
   }
+
+  /** Load reference data codes for given domain */
+  getReferenceCodes(domain: string): Promise<ReferenceCode[]> {
+    function parseDates(referenceCode: DatesAsStrings<ReferenceCode>): ReferenceCode {
+      return {
+        ...referenceCode,
+        expiredDate: referenceCode.expiredDate && new Date(referenceCode.expiredDate),
+        subCodes: (referenceCode.subCodes ?? []).map(parseDates),
+      }
+    }
+
+    return this.get<DatesAsStrings<ReferenceCode[]>>({
+      path: `/api/reference-domains/domains/${encodeURIComponent(domain)}/codes`,
+    }).then(codes => codes.map(parseDates).sort((code1, code2) => (code1.listSeq ?? 0) - (code2.listSeq ?? 0)))
```

Comment:
isn't `listSeq` always there 🤔?

Reply:
no. lol

Reply:
😬 - in that case should the `ReferenceCode` interface reflect that? e.g. `listSeq?: number`